### PR TITLE
VEIOSS-66 | Allow Trader to View Previous Returns

### PR DESCRIPTION
**Pull request by warrenskiwork1979**

### CODE REVIEW

**Comment by warrenskiwork1979** [comment]
retest this please

File: app/models/etmp/EtmpVatReturnGoodsSupplied.scala [hunk 1]
```diff
@@ -16,15 +16,16 @@
 
 package models.etmp
 
-import play.api.libs.json.{Format, Json}
+import play.api.libs.json.{Json, OFormat}
 
 case class EtmpVatReturnGoodsSupplied(
-                                       msOfConsumption: String,
+                                       msOfConsumption: String, // TODO country?
```
Comment (DevAndyroo) [inline app/models/etmp/EtmpVatReturnGoodsSupplied.scala:22]: I think this can be removed

File: app/connectors/VatReturnConnector.scala [hunk 2]
```diff
@@ -38,6 +39,9 @@ class VatReturnConnector @Inject()(config: Configuration, httpClient: HttpClient
       s"$baseUrl/return",
       coreVatReturn
     )
+  }
 
+  def get(period: Period)(implicit hc: HeaderCarrier): Future[EtmpVatReturn] = { // TODO - HttpParser
```
Comment (DevAndyroo) [inline app/connectors/VatReturnConnector.scala:44]: Is the HttpParser needed?

File: app/viewmodels/previousReturns/SubmittedReturnForPeriodSummary.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    val value = etmpVatReturn.returnVersion
+
+    SummaryListRowViewModel(
+      key = "submittedReturnForPeriod.summary.submittedDate",
+      value = ValueViewModel(HtmlContent(value.format(etmpDateFormatter))).withCssClass("govuk-table__cell--numeric govuk-!-padding-right-0")
+    )
+  }
+
+  def rowPaymentDueDate(period: Period)(implicit messages: Messages): SummaryListRow = {
+    // TODO Payment due by -> period.paymentDeadline when merged? -> Will need to covert from LocalDate to LocalDateTime
```
Comment (DevAndyroo) [inline app/viewmodels/previousReturns/SubmittedReturnForPeriodSummary.scala:72]: I think paymentDeadline should be there now. Why does it need to be a DateTime though?

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/connectors/VatReturnConnector.scala (hunk 2 above)
Comment (warrenskiwork1979) [reply app/connectors/VatReturnConnector.scala:44]: Was just thinking or error handling at this point. Do we not need it? I think I've already added it in 167.

File: app/models/etmp/EtmpVatReturnGoodsSupplied.scala (hunk 1 above)
Comment (warrenskiwork1979) [reply app/models/etmp/EtmpVatReturnGoodsSupplied.scala:22]: OK will remove

File: app/viewmodels/previousReturns/SubmittedReturnForPeriodSummary.scala (hunk 3 above)
Comment (warrenskiwork1979) [reply app/viewmodels/previousReturns/SubmittedReturnForPeriodSummary.scala:72]: Ignore that, that's when I was initially passing in  vat return model as wasn't sure how deadline was being calculated. Will remove and add deadline

**Review by DevAndyroo — APPROVED** [review]

**Pull request merged**